kernel/kernel-defaults/src/test/java/io/delta/kernel/defaults/benchmarks/WorkloadBenchmark.java

```diff
@@ -0,0 +1,120 @@
+ * Generic JMH benchmark for all workload types. Automatically loads and runs benchmarks based on
+ * JSON workload specifications.
+ */
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@Fork(value = 1, warmups = 1)
+@Warmup(iterations = 3, time = 1)
+@Measurement(iterations = 5, time = 1)
+public class WorkloadBenchmark {
+
+  /**
+   * Dynamic benchmark state that reads the workload specification and sets up the workload runner
+   */
+  @State(Scope.Thread)
+  public static class BenchmarkState {
+    // This parameter will be set dynamically in the main method to include all workload spec paths.
+    @Param({})
+    private String workloadPath;
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+    private WorkloadSpec workload;
+    private WorkloadRunner runner;
+
+    /**
+     * Setup method that runs before each benchmark invocation. Reads the workload specification
+     * from the given path and initializes the corresponding workload runner.
+     *
+     * @throws Exception If any error occurs during setup.
+     */
+    @Setup(Level.Invocation)
+    public void setup() throws Exception {
+      workload = objectMapper.readValue(new File(workloadPath), WorkloadSpec.class);
+      runner = workload.getRunner(RESOURCES_DIR);
+    }
+
+    /** @return The workload specification for this benchmark invocation. */
+    public WorkloadSpec getWorkloadSpecification() {
+      return workload;
+    }
+
+    /** @return The workload runner initialized for this benchmark invocation. */
+    public WorkloadRunner getRunner() {
+      return runner;
+    }
+  }
+
+  /**
+   * Benchmark method that executes the workload specified in the state.
+   *
+   * @param state The benchmark state containing the workload specification and runner.
+   * @param blackhole The Blackhole to consume results and prevent dead code elimination.
+   * @throws Exception If any error occurs during workload execution.
+   */
+  @Benchmark
+  public void benchmarkWorkload(BenchmarkState state, Blackhole blackhole) throws Exception {
+    WorkloadRunner workload = state.getRunner();
+    workload.executeAsBenchmark(blackhole);
+  }
+
+  public static void main(String[] args) throws RunnerException {
+    // Get workload specs from the workloads directory
+    List<String> workloadSpecPaths = BenchmarkUtils.loadAllWorkloads(WORKLOAD_SPECS_DIR);
+    if (workloadSpecPaths.isEmpty()) {
+      throw new RunnerException(
+          "No workloads found. Please add workload specs to the workloads directory.");
+    }
+
+    // Convert paths into a String array for JMH. JMH requires that parameters be of type String[].
+    String[] paths = workloadSpecPaths.toArray(new String[0]);
+
+    // Create and run the benchmark
+    Options opt =
+        new OptionsBuilder()
+            .include(WorkloadBenchmark.class.getSimpleName())
+            .param("workloadPath", paths)
+            .forks(1)
+            .warmupIterations(3)
```

Comment:
should all of these configurations eventually be configurable? // doesn't need to be done in this PR, just curious. if so, can we add this as a TODO like so: `TODO: [delta-io/delta#12345] Allow users to configure XXX` and can we also track this future work in a tracking github issue?

Reply:
Good point. In fact there's a lot of mental TODO's that I haven't formally put into Issues/comments. I'll add that.

Reply:
Still needs to be filled out fully, but here it is: https://github.com/delta-io/delta/pull/5139